src/code/ContainerRegistryServerAPICalls.cs

```diff
@@ -1705,12 +1720,63 @@ private string PrependMARPrefix(string packageName)
 
             // If the repostitory is MAR and its not a wildcard search, we need to prefix the package name with MAR prefix.
             string updatedPackageName = Repository.IsMARRepository() && packageName.Trim() != "*"
-                                            ? string.Concat(prefix, packageName)
+                                            ? packageName.StartsWith(prefix) ? packageName : string.Concat(prefix, packageName)
                                             : packageName;
 
             return updatedPackageName;
         }
 
+        private FindResults FindPackages(string packageName, bool includePrerelease, out ErrorRecord errRecord)
+        {
+            _cmdletPassedIn.WriteDebug("In ContainerRegistryServerAPICalls::FindAll()");
+            errRecord = null;
+            string containerRegistryAccessToken = GetContainerRegistryAccessToken(out errRecord);
+            if (errRecord != null)
+            {
+                return emptyResponseResults;
+            }
+
+            var pkgResult = FindAllRepositories(containerRegistryAccessToken, out errRecord);
+            if (errRecord != null)
+            {
+                return emptyResponseResults;
+            }
+
+            List<Hashtable> repositoriesList = new List<Hashtable>();
+            var isMAR = Repository.IsMARRepository();
+
+            // Convert the list of repositories to a list of hashtables
+            foreach (var repository in pkgResult["repositories"].ToList())
+            {
+                string repositoryName = repository.ToString();
+
+                if (isMAR && !repositoryName.StartsWith("psresource/"))
```

Comment:
it would be better to use a variable for the prefix here, like `PSRepositoryInfo.MARPrefix` here, or do you need to check `InternalHooks.MARPrefix` as well to determine the prefix like done in the PrependMARPrefix() method?

```suggestion
                if (isMAR && !repositoryName.StartsWith(PSRepositoryInfo.MARPrefix))
```